### PR TITLE
[3.x] feat: allow to customise the signature_method

### PR DESCRIPTION
**Pull request by ankurk91**
Hi,

Vonage allows us to customise the signature method (algo) through their dashboard (see screenshot), the default value is set to `HMAC_SHA256`.

This package was hard coded to use `md5hash`, this PR introduce a new config key `signature_method` which allows us to override the parameter.
Web capture_30-4-2023_10815_dashboard nexmo com

Vonage uses this algorithm to sign the incoming webhooks as well as outgoing SMS via the SDK.

PS.
 I have kept backward compatibility, the default algo in sdk is still the same.

### CODE REVIEW

File: src/VonageChannelServiceProvider.php [hunk 1]
```diff
@@ -22,8 +22,6 @@ public function register()
         $this->app->singleton(Client::class, function ($app) {
             $config = $app['config']['vonage'];
 
-            $httpClient = null;
```
Comment (ankurk91) [inline src/VonageChannelServiceProvider.php]: This line was unused always since the variable is being initialised in the very next line.

This change is not related to the new feature introduced .

**Review by driesvints — APPROVED** [review]
